### PR TITLE
refactor(host): extract Beads repository lifecycle manager

### PR DESCRIPTION
**Pull request by Maxsky5**
Summary
This PR extracts Beads CLI lifecycle responsibilities out of `createBeadsTaskRepository` so the repository can focus on task operations and cache invalidation.

Goal
`createBeadsTaskRepository` previously owned CLI context resolution, shared Dolt server ownership and shutdown, command-runner binding, task-list caching, task operations, and error mapping in one large adapter factory. The goal is to separate Beads process lifecycle behavior from task mutation/read behavior, making shutdown semantics and command binding easier to reason about and test independently.

Technical choices
- Added `BeadsCliContextManager` to own Beads context request resolution, shared-server context flights, failed-flight eviction before retry, owned shared Dolt server tracking, and close-time shutdown.
- Added `BdCommandProvider` to bind default `bd` and `bd --json` runners to a repo-scoped shared-server context while preserving configured runner behavior without pre-resolving context.
- Kept `createBeadsTaskRepository` focused on task operations, task-list cache invalidation, diagnostics wiring, and error mapping.
- Moved shared Beads adapter fixtures into focused test support. Lifecycle behavior now lives in the manager tests; repository tests keep only a narrow close-delegation seam plus task operation behavior.
- Per the cleanup pass, removed unused provider surface and duplicate repository-level lifecycle coverage.

Verification
- No `Cargo.toml` exists in this checkout, so cargo checks do not apply.
- `bun run lint` passed. It still reports existing optional-chain warnings outside this Beads scope, but exits successfully.
- `bun run typecheck` passed.
- `bun run test` passed.
- `bun run build` passed. Build emits existing Vite/chunk-size warnings, but exits successfully.

## Summary by CodeRabbit

* **Bug Fixes & Improvements**
  * Enhanced concurrent request handling and shared server lifecycle management
  * Improved error recovery for context resolution failures
  * Better resource cleanup and graceful shutdown

### CODE REVIEW

File: packages/host/src/adapters/beads/beads-cli-context-manager.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                readyCliContexts.delete(request.cacheKey);
+              }
+            }),
+            flight: reservation.flight,
+            rememberOwnedContext,
+            resolveContext: resolveCliContext(request.repoPath, sharedServerOptions),
+          });
+          return yield* restore(awaitBeadsCliContextFlight(reservation.flight));
+        }),
+      );
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/host/src/adapters/beads/beads-cli-context-manager.ts:192]: Fixed in 5b5aee2bb. The manager now re-checks closing at reservation time inside the uninterruptible sections for both branches, and the non-shared resolver is lazy behind that guard. I kept this to a direct guard rather than an active resolver registry because it closes the race without adding new lifecycle state. Added regression tests covering no new non-shared flight or shared reservation after close begins.

**Comment by Maxsky5** [comment]
Review follow-up pushed in 5b5aee2bb.

Fixed:
- Added close-time reservation guards in BeadsCliContextManager for both non-shared context flights and shared ready-context reservations.
- Made the non-shared raw context resolver lazy behind the reservation guard so it cannot start after close begins.
- Added regression coverage for both close-started reservation paths.
- Added runBdForRepo coverage for default non-JSON binding and configured runBd bypass/no pre-resolution behavior.
- Made repo-bound runner signatures explicit and documented the raw runBdJson runner shape.
- Tightened the createTestBeadsTaskRepository toolDiscovery seam and added fake bd temp directory cleanup.

Not changed:
- Kept the conditional resolveCliContext spread into createBeadsCliContextManager because it intentionally omits absent optional dependencies and matches the surrounding factory style.
- Kept mapTaskStoreErrors on close() so all public repository methods keep the same boundary error-mapping behavior.
- Did not add a bare close-without-diagnostics repository test; that would only assert an empty lifecycle case, while manager tests now cover shutdown behavior and the repository test still uses a real production diagnostics path to register an owned context before close.
- Did not convert test-support filesystem helpers to Effect. They are test fixture setup around Node filesystem APIs, not production host ports or application services, and converting them would add broad test churn without improving the lifecycle extraction.
- React Doctor, walkthrough, rate-limit, and summary-only comments are informational or duplicates of the fixed items above.

Verification:
- bun test packages/host/src/adapters/beads/beads-cli-context-manager.test.ts packages/host/src/adapters/beads/bd-command-provider.test.ts packages/host/src/adapters/beads/beads-task-repository.test.ts
- bun run lint
- bun run typecheck
- bun run test
- bun run build
- bun run --filter @openducktor/host lint
